music_assistant/providers/chromecast/__init__.py

```diff
@@ -624,19 +624,47 @@ async def _disconnect_chromecast(self, castplayer: CastPlayer) -> None:
         castplayer.status_listener = None
         self.castplayers.pop(castplayer.player_id, None)
 
+    def _clamp_image_size(self, limit: int, orig_url: str | None) -> str | None:
+        """Clamp image for media item."""
+        if not orig_url or limit == 0:
+            return orig_url
+
+        try:
+            from urllib.parse import parse_qsl, urlencode, urlparse
+
+            url = urlparse(orig_url)
+            qsl = dict(parse_qsl(url.query))
+
+            size = int(qsl.get("size", "0"))
+            if size != 0 and size <= limit:
+                return orig_url
+
+            qsl["size"] = str(limit)
+            url = url._replace(query=urlencode(qsl))
+            return url.geturl()
+        except Exception as err:
+            self.logger.warning(
+                "Error while clamping image url size: %s",
+                str(err),
+                exc_info=err if self.logger.isEnabledFor(10) else None,
+            )
+
+        return orig_url
```

Comment:
This assumes that every image has a query string with a size parameter while that is strictly speaking not true. If you want to absolutely guarantee the image size, you will have to request the image through the image proxy, which can also do resizing. 

I'm ok with this change as a start but you will still run into the issue if you receive an URL to an image that has no size in a query string but is just a super large image

Reply:
If you would prefer I could make the change to the call to `get_image_url` in the `PlayerQueuesController` instead, that would catch all (?) URL generation but would be affecting all players not just Chromecast.

Reply:
Done!

I changed the calls to `get_image_url` in `PlayerQueuesController` and the Chromecast player.

I didn't touch the calls in the airplay player (already sets size=500) or slimproto + sonos players (no known issues there). They will get the 512 limit from `PlayerQueuesController` but not in their own calls, that seems a bit inconsistent -- would it be better to add the 512 to those two as well while I'm here?

Reply:
yeah, and always feel free to submit/propose optimizations when you see fit.
Thanks!